### ISSUE TITLE
Intermittent - TypeError: list indices must be integers or slices, not str

### ISSUE THREAD

**Issue by SmitaPatankar** [cannatag/ldap3#854]
**Steps to reproduce:**
```
pip install ldap3==2.8
vi test.py
```
```
#!/usr/bin/env python3
import ldap3
import ssl
SERVER = ""
USER=''
PASSWORD=''
BASE_DN=""
tls = ldap3.Tls(validate=ssl.CERT_NONE, version=ssl.PROTOCOL_TLSv1_2)
server = ldap3.Server(SERVER, get_info=ldap3.ALL, mode=ldap3.IP_V4_PREFERRED, tls=tls, use_ssl=True)
with ldap3.Connection(
    server,
    authentication=ldap3.NTLM,
    auto_bind=True,
    password=PASSWORD,
    read_only=True,
    receive_timeout=7200,
    user=USER,
    raise_exceptions=False
) as ldap_connection:
    query="(&(objectclass=group)(objectcategory=group)(distinguishedName=*))"
    attributes=["*"]
    ldap_connection.search(search_base=BASE_DN, search_filter=query, search_scope=ldap3.SUBTREE, attributes=attributes, get_operational_attributes=True)
    print(ldap_connection.entries)
```
```
python test.py
```

**Expected Result:**
Group entries should be displayed

**Actual Result:**
```
Traceback (most recent call last):
  File "test.py", line n, in <module>
    ldap_connection.search(search_base=BASE_DN, search_filter=query, search_scope=ldap3.SUBTREE, attributes=attributes, get_operational_attributes=True)
  File "/opt/myproject/venv/lib/python3.6/site-packages/ldap3/core/connection.py", line 839, in search
    response = self.post_send_search(self.send('searchRequest', request, controls))
  File "/opt/myproject/venv/lib/python3.6/site-packages/ldap3/strategy/sync.py", line 139, in post_send_search
    responses, result = self.get_response(message_id)
  File "/opt/myproject/venv/lib/python3.6/site-packages/ldap3/strategy/base.py", line 411, in get_response
    if self.do_search_on_auto_range(self._outstanding[message_id], response):
  File "/opt/myproject/venv/lib/python3.6/site-packages/ldap3/strategy/base.py", line 736, in do_search_on_auto_range
    self.do_next_range_search(request, resp, attr_name)
  File "/opt/myproject/venv/lib/python3.6/site-packages/ldap3/strategy/base.py", line 689, in do_next_range_search
    response['raw_attributes'][attr_type] += current_response['raw_attributes'][attr_name]
TypeError: list indices must be integers or slices, not str
```

**Comment by BlackVoid** [comment]
I'm getting this issue when doing queries towards AD, after some debugging I found the change that causes the issue: https://github.com/cannatag/ldap3/commit/72edc5eead0682e1bd787e053018c64079cd88e5#diff-b934c360851c1cfbfae4ea478912b14aL696-L703

**Comment by cannatag** [comment]
Please, check code in dev. 

Thanks,
Giovanni

**Comment by cannatag** [comment]
Fixed in 2.8.1
